lldb/include/lldb/Target/ExecutionContext.h

```diff
@@ -566,6 +559,53 @@ class ExecutionContext {
   lldb::StackFrameSP m_frame_sp; ///< The stack frame in thread.
 };
 
+/// A wrapper class representing an execution context with non-null Target
+/// and Process pointers, a locked API mutex and a locked ProcessRunLock.
+/// The locks are private by design: to unlock them, destroy the
+/// CompleteExecutionContext.
+struct CompleteExecutionContext : ExecutionContext {
```

Comment:
[bikeshedding]
Should this be called `StoppedExecutionContext`? My reasoning is that it would both emphasize that the process is stopped (and will remain) stopped, which automatically implies that you have a process.
[/bikeshedding]

Reply:
Ohhh, so shadow those methods and mark them as deprecated; interesting idea, I'll give it a try on the next pr for sure.